src/amd/compiler/aco_lower_bool_phis.cpp

```diff
@@ -206,35 +207,32 @@ void lower_linear_bool_phi(Program *program, Block *block, aco_ptr<Instruction>&
 void lower_bool_phis(Program* program)
 {
    for (Block& block : program->blocks) {
-      std::vector<aco_ptr<Instruction>> instructions;
-      std::vector<aco_ptr<Instruction>> non_phi;
-      instructions.swap(block.instructions);
-      block.instructions.reserve(instructions.size());
-      unsigned i = 0;
-      for (; i < instructions.size(); i++)
-      {
-         aco_ptr<Instruction>& phi = instructions[i];
-         if (phi->opcode != aco_opcode::p_phi && phi->opcode != aco_opcode::p_linear_phi)
-            break;
-         if (phi->opcode == aco_opcode::p_phi && phi->definitions[0].regClass() == s2) {
-            non_phi.emplace_back(std::move(lower_divergent_bool_phi(program, &block, phi)));
-         } else if (phi->opcode == aco_opcode::p_linear_phi && phi->definitions[0].regClass() == s1) {
+      // find the first non linear phi
+      auto IsNonlinearPhi = [] (const aco_ptr<Instruction>& phi) -> bool {
+         return phi->opcode != aco_opcode::p_phi && phi->opcode != aco_opcode::p_linear_phi;
+      };
+      auto end = std::find_if(block.instructions.begin(), block.instructions.end(), IsNonlinearPhi);
+
+      // partition divergent phi's to the back
+      auto IsNotDivergentPhi = [] (const aco_ptr<Instruction>& phi) -> bool {
```

Comment:
This searches for divergent _boolean_ phis, should be mentioned in the comment at least (the variable name would get a bit long, though °° )

Reply:
Naming is hard. I will update the comment